lib/logging/post_auth.rb

```diff
@@ -1,28 +1,25 @@
 module Logging
   class PostAuth
     def execute(params:)
-      authentication_result = params.fetch(:authentication_result)
-
-      if authentication_result == 'Access-Accept'
-        handle_access_accept(params)
-        return true
-      elsif authentication_result == 'Access-Reject'
-        return true
-      end
-
-      false
+      valid_request?(params) ? handle_access_request(params) : false
     end
 
   private
 
     VALID_MAC_LENGTH = 17
 
-    def handle_access_accept(params)
+    def valid_request?(params)
+      valid_auth_results = ['Access-Accept', 'Access-Reject']
+      auth_result = params.fetch(:authentication_result)
+      valid_auth_results.include?(auth_result)
+    end
+
+    def handle_access_request(params)
       username = params.fetch(:username)
-      return if username == 'HEALTH'
+      return true if username == 'HEALTH'
 
-      create_session(params)
       update_user_last_login(username)
```

Comment:
Probably don't want to update user last login if result was an Access-Reject

Reply:
Cool - fixed this in https://github.com/alphagov/govwifi-logging-api/pull/59/commits/8d149571da385b35dfa8f0a6446de835f9b19633#diff-7ff9309fec8ca53f84ec3f487e050597R9